chrome/content/zotero/xpcom/utilities_internal.js

```diff
@@ -360,6 +360,35 @@ Zotero.Utilities.Internal = {
 	},
 	
 	
+	/**
+	 * Decode a binary string into a typed Uint8Array
+	 *
+	 * @param {String} data - Binary string to decode
+	 * @return {Uint8Array} Typed array holding data
+	 */
+	_decodeToUint8Array: function (data) {
+		var buf = new ArrayBuffer(data.length);
+		var bufView = new Uint8Array(buf);
+		for (let i = 0; i < data.length; i++) {
+			bufView[i] = data.charCodeAt(i);
+		}
+		return bufView;
+	},
+	
+	
+	/**
+	 * Decode a binary string to UTF-8 string
+	 *
+	 * @param {String} data - Binary string to decode
+	 * @return {String} UTF-8 encoded string
+	 */
+	decodeToUTF8: function (data) {
```

Comment:
I think `decodeUTF8` would be more accurate. It's not really decoding _to_ UTF-8 — the bytes are UTF-8-encoded text, and we're decoding that to a JavaScript string.